Document glue job module (#70)
LBHackney-IT/Data-Platform-Playbook: docs/playbook/create-glue-jobs.md

---
title: Creating Glue jobs
description: "Creating Glue jobs in terraform"
layout: playbook_js
tags: playbook
---

## Creating Glue jobs using the Glue jobs Terraform template

#### 1. Copy an existing Glue job or use the example below and populate the variables
If the file `23-aws-glue-jobs-<YOUR_DEPARTMENT_NAME>.tf` in the terraform directory exists, copy an existing module block. 

> 

Reviewer: Explain how to do this using the Github UI.

- Do not change any spacing before the `=`
- Do not remove the `"quotation marks"` around any values which use them

_Create a new tf file for your department, named `23-aws-glue-jobs-<YOUR_DEPARTMENT_NAME>.tf` if it does not exist and refer to the [example](#example-module-block) below_

Update the `module` name in the format `your-unique-module-name`, and unique to all other module names in this file (it is helpful to keep the same naming convention as your dataset/folder)

#### The following input variables are required:
- __department__: This will be `module.department_<YOUR_DEPARTMENT_NAME>`
- __job_name__: The name of your glue job prefixed with your department name e.g. "housing repairs address cleaning"
- __glue_scripts_bucket_id__: `module.glue_scripts.bucket_id`

#### The following variables are optional:
- __job_description__: A description of the AWS glue job e.g. "Glue job description"
- __job_parameters__: Parameters used in your Glue job. For example the source database and table
    - If you are setting __job_parameters__, you should include the below parameter:

>       - ```

Reviewer: ```suggestion
      ```
```

>         "--TempDir" = "${module.glue_temp_storage.bucket_url}/${module.department_housing_repairs.identifier}/"

Reviewer: If we want everyone to include this can we just accept the temp bucket ID as a variable and get the module to add this as a variable? It feels simpler to me

Reviewer 2: Yes sounds like a good idea

        ```
- __script_name__: Name of the Glue job script. If no value is provided, then this will default to the same name as the job name
- __workflow_name__: Workflow to add the triggers to.
- __triggered_by_crawler__: Can populate either this variable, the __triggered_by_job__ variable or the schedule.
  The job created in this module will be triggered on completion of either
  the crawler given here or the job given in job_to_trigger or the schedule.
- __triggered_by_job__: Can populate either this variable, the __triggered_by_crawler__ variable or the schedule.
  The job created in this module will be triggered on completion of either
  the job given here or the crawler given in __triggered_by_crawler__.

> - __crawler_details__: Inputs required to create a crawler. These are:

Reviewer: ```suggestion
- __crawler_details__: Inputs required to create a crawler. If this is not provided then no crawler will be created. These are:
```

    - __database_name__: Glue database where results are written. 
    - __s3_target_location__: The S3 location to be crawled.
    - __table_prefix__ (Optional): The table prefix used for catalog tables that are created.
    - __configuration__ (Optional): By default, the `TableGroupingPolicy` will be set to `CombineCompatibleSchemas`  

> - __schedule__: Schedule to run the Glue job. Can populate either this variable, __triggered_by_job__ or the __triggered_by_crawler__. 

Reviewer: Worth mentioning that this is expecting a cron and maybe linking to the AWS cron docs

- __number_of_workers_for_glue_job__: Specify the number of workers to use for the glue job. The is set to 2 by default.
- __glue_job_worker_type__: Worker type for Glue job. The default is Standard. This can either be set as "Standard", "G.1X" or "G.2X". 
- __max_concurrent_runs_of_glue_job__: Max number of concurrent runs for the glue job. The is set to 1 by default.

> - __trigger_enabled__: Set to "false" to disable scheduled or conditional triggers for the glue job. This is set to `true` by default.

Reviewer 2: ```suggestion
- __trigger_enabled__: Set to `false` to disable scheduled or conditional triggers for the glue job. This is set to `true` by default.
```

> #### 3. Committing your changes: The Data Platform team needs to approve any changes to the code, so your change won't happen automatically. To submit your change:

Reviewer: ```suggestion
#### 3. Committing your changes using the GitHub UI: The Data Platform team needs to approve any changes to the code, so your change won't happen automatically. To submit your change:
```

>   - Provide a description to explain what you've changed

Reviewer: ```suggestion
  - Provide a name and description to explain what you've changed
```

  - Select the option to create a `new branch` for this commit (i.e. the code you've changed). You can just use the suggested name for your branch.

>   - Once you click 'Propose changes' you'll have the opportunity to add even more detail if needed before submitted for review.

Reviewer: ```suggestion
  - Once you click 'Propose changes' you'll have the opportunity to add even more detail if needed before submitting it for review.  Click 'Create Pull Request' when finished.
```

>   - You'll receive an email to confirm that your changes have been approved.

Reviewer: ```suggestion
  - You'll receive an email to confirm when your changes have been approved.
```
